### PR TITLE
Test: Added Integration test for Work Status Controller

### PR DESCRIPTION
**Pull request by aagusuab**
### Description of your changes

Fixes #

I have:

- [x] Read and followed Caravel's [Code of conduct](https://github.com/Azure/k8s-work-api/blob/master/code-of-conduct.md).
- [x] Run `make reviewable` to ensure this PR is ready for review.

### CODE REVIEW

File: pkg/controllers/suite_test.go [hunk 1]
```diff
@@ -39,11 +40,12 @@ import (
 // These tests use Ginkgo (BDD-style Go testing framework). Refer to
 // http://onsi.github.io/ginkgo/ to learn more about Ginkgo.
 var (
-	cfg        *rest.Config
-	k8sClient  kubernetes.Interface
-	workClient workclient.Interface
-	testEnv    *envtest.Environment
-	setupLog   = ctrl.Log.WithName("test")
+	cfg           *rest.Config
+	k8sClient     kubernetes.Interface
```
Comment (Ealianis) [inline pkg/controllers/suite_test.go:44]: #### Suggestion > Revision
For storytelling purposes let's be sure to indicate what cluster these clients are for. Prefix them with hub or spoke.

File: pkg/controllers/work_status_controller_integration_test.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilrand "k8s.io/apimachinery/pkg/util/rand"
+	workv1alpha1 "sigs.k8s.io/work-api/pkg/apis/v1alpha1"
+)
+
+var _ = Describe("Work Status Controller", func() {
```
Comment (Ealianis) [inline pkg/controllers/work_status_controller_integration_test.go:34]: #### Suggestion > Revision
The Describe / Context / It don't tell the correct story. A few asserts  (IT blocks) are missing.

You have:

- Describe = Work Status Controller
  - Context = A Work object with manifests has been created
    - It = Resource was removed from the Work Status
    - It = Resource is deleted from the AppliedResources of the AppliedWork
    
  It should be more like:
 
  - Describe = Work Status Reconciler
  - Context = Receives a request where where a Work's ManifestConditions does not contain the metadata of an existing AppliedResourceMeta.
   It = Should delete the resource from the spoke cluster.
  - Context = Receives a request where where a Work ManifestCondition exists but there is not a respective AppliedResourceMeta.
    - It = Should update the AppliedWork's ApplieResourceMeta to contain the resource meta .

Comment (aagusuab) [reply pkg/controllers/work_status_controller_integration_test.go:34]: I have edited the integration test description to be similar to the above suggestion, more concise and fitting the integration test contexts.

File: pkg/controllers/work_status_controller_integration_test.go [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+			Expect(err).ToNot(HaveOccurred())
+
+			Eventually(func() bool {
+				gvr := schema.GroupVersionResource{
+					Group:    "core",
+					Version:  "v1",
+					Resource: "ConfigMap",
+				}
+				_, err := dynamicClient.Resource(gvr).Namespace(resourceNamespace).Get(context.Background(), resourceName, metav1.GetOptions{})
+				return err != nil
```
[the remark this answers is not in the document]
Comment (aagusuab) [reply pkg/controllers/work_status_controller_integration_test.go:130]: Added new lines to all of the affected return statements.

File: pkg/controllers/suite_test.go (hunk 1 above)
Comment (aagusuab) [reply pkg/controllers/suite_test.go:45]: Since there were more changes to be affected and not just the test for this PR, I have added a todo.

Comment (Ealianis) [reply pkg/controllers/suite_test.go:45]: @aagusuab , what other changes, it's just a prefix to the var name.

Comment (Ealianis) [reply pkg/controllers/suite_test.go:45]: Nm, I see what the issue is from your below comment.

File: pkg/controllers/work_status_controller_integration_test.go [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+		Expect(createWorkErr).ToNot(HaveOccurred())
+
+		Eventually(func() bool {
+			appliedWorkObject, _ := workClient.MulticlusterV1alpha1().AppliedWorks().Get(context.Background(), workName, metav1.GetOptions{})
+
+			return appliedWorkObject.Spec.WorkName == workName
+		}, timeout, interval).Should(BeTrue())
+	})
+
+	AfterEach(func() {
```
Comment (Ealianis) [inline pkg/controllers/work_status_controller_integration_test.go:106]: #### Comment > General
Be aware that not cleaning up the setup done in AfterEach is only safe due to the randomness of the object meta used.  We should clean up all resources between tests.

**Review by Ealianis — APPROVED** [review]

File: pkg/controllers/work_status_controller_integration_test.go [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+		err := k8sClient.CoreV1().Namespaces().Delete(context.Background(), workNamespace, metav1.DeleteOptions{})
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	Context("Receives a request where a Work's manifest condition does not contain the metadata of an"+
+		"existing AppliedResourceMeta", func() {
```
[the remark this answers is not in the document]
Comment (aagusuab) [reply pkg/controllers/work_status_controller_integration_test.go:112]: I reverted this, Keeping everything similar.

File: pkg/controllers/work_status_controller_integration_test.go (hunk 4 above)
Comment (aagusuab) [reply pkg/controllers/work_status_controller_integration_test.go:106]: Added TODO for checking the clean up of resources.

**Pull request merged**